tests/extension/oneapi_kernel_compiler_spirv/kernel_compiler_spirv.cpp

```diff
@@ -0,0 +1,235 @@
+
+  sycl::buffer a_buffer(&a, sycl::range<1>(1));
+
+  T* const b_ptr = sycl::malloc_shared<T>(1, q);
+  b_ptr[0] = b;
+
+  T output{};
+  sycl::buffer output_buffer(&output, sycl::range<1>(1));
+
+  q.submit([&](sycl::handler& cgh) {
+    sycl::local_accessor<T, 1> local(1, cgh);
+    // Pass T for scalar parameter.
+    cgh.set_arg(0, a);
+    // Pass USM pointer for OpTypePointer(CrossWorkgroup) parameter.
+    cgh.set_arg(1, b_ptr);
+    // Pass sycl::accessor for OpTypePointer(CrossWorkgroup) parameter.
+    cgh.set_arg(
+        2, output_buffer.template get_access<sycl::access::mode::write>(cgh));
+    // Pass sycl::local_accessor for OpTypePointer(Workgroup) parameter.
+    cgh.set_arg(3, local);
+    cgh.parallel_for(sycl::range<1>{1}, kernel);
+  });
+
+  sycl::host_accessor out{output_buffer};
+  CHECK(out[0] == expected);
+  sycl::free(b_ptr, q);
+}
+
+void testStruct(sycl::queue& q, const sycl::kernel& kernel) {
+  const auto num_args = kernel.get_info<sycl::info::kernel::num_args>();
+  REQUIRE(num_args == 2);
+
+  // This definition must match the one used in the kernel.
+  struct S {
+    std::int32_t i;
+    cl_float f;
+    std::int32_t* p;
+    struct Inner {
+      std::int32_t i;
+      float f;
+      std::int32_t* p;
+    } inner;
+  };
+
+  // Any constants can be used to initialize this input.
+  std::int32_t* const in_p0 = sycl::malloc_shared<std::int32_t>(1, q);
+  std::int32_t* const in_p1 = sycl::malloc_shared<std::int32_t>(1, q);
+  *in_p0 = 3;
+  *in_p1 = 6;
+  S input{1, 2.0f, in_p0, S::Inner{4, 5.0f, in_p1}};
+
+  std::int32_t* const out_p0 = sycl::malloc_shared<std::int32_t>(1, q);
+  std::int32_t* const out_p1 = sycl::malloc_shared<std::int32_t>(1, q);
+  *out_p0 = 0;
+  *out_p1 = 0;
+  S* output = sycl::malloc_shared<S>(1, q);
+  *output = S{0, 0, out_p0, S::Inner{0, 0, out_p1}};
+
+  q.submit([&](sycl::handler& cgh) {
+     cgh.set_arg(0, input);
+     cgh.set_arg(1, output);
+     cgh.parallel_for(sycl::range<1>{1}, kernel);
+   }).wait();
+
+  // For each scalar struct member, output == (2 * input). For pointer members,
+  // *output == (2 * (*input)).
+  CHECK(output->i == input.i * 2);
+  CHECK(output->f == input.f * 2);
+  CHECK(*output->p == (*input.p) * 2);
+  CHECK(output->inner.i == input.inner.i * 2);
+  CHECK(output->inner.f == input.inner.f * 2);
+  CHECK(*output->inner.p == (*input.inner.p) * 2);
+
+  sycl::free(output, q);
+  sycl::free(in_p0, q);
+  sycl::free(in_p1, q);
+  sycl::free(out_p0, q);
+  sycl::free(out_p1, q);
+}
+
+void testKernelsFromSpvFile(std::string kernels_file,
+                            std::string fp16_kernel_file,
+                            std::string fp64_kernel_file) {
+  const auto getKernel =
+      [](sycl::kernel_bundle<sycl::bundle_state::executable>& bundle,
+         const std::string& name) {
+        return bundle.ext_oneapi_get_kernel(name);
+      };
+
+  sycl::device d;
+  sycl::context ctx{d};
+  sycl::queue q{ctx, d};
```

Comment:
Is there any fundamental reason to use a default device and a context instead of just the default queue?